src/tools/sign-oci-layer-root-hashes/src/registry.rs

```diff
@@ -406,19 +409,77 @@ fn decompress_file(compressed_path: &Path, decompressed_path: &Path) -> Result<(
     Ok(())
 }
 
-fn attach_tarfs_index(path: &Path) -> Result<()> {
-    info!("Adding tarfs index to layer");
-    let mut file = std::fs::OpenOptions::new()
-        .read(true)
+fn attach_erofs_meta(path: &Path) -> Result<()> {
+    info!("Creating erofs metadata. Appending decompressed tar to erofs metadata");
+
+    // Create an erofs image using mkfs.erofs
+    let erofs_path = path.with_extension("erofs");
+    debug!(
+        "Creating erofs metadata {:?} from {:?}",
+        &erofs_path, &path
+    );
+
+    let status = Command::new("mkfs.erofs")
+        .args([
+            "--tar=i",
+            "-T", "0", // zero out unix time
+            "--mkfs-time", // clear out mkfs time in superblock, but keep per-inode mtime
+            "-U", EROFS_METADATA_UUID, // set UUID to something specific
+            "--aufs", // needed to convert OCI whiteouts/opaque to overlayfs metadata
+            "--quiet",
+            erofs_path.to_str().unwrap(),
+            path.to_str().unwrap(),
+        ])
+        .status()
+        .context("Failed to execute mkfs.erofs command")?;
+
+    if !status.success() {
+        return Err(anyhow!(
+            "mkfs.erofs failed with status: {}",
+            status.code().unwrap_or(-1)
+        ));
+    }
+
+    // Append the decompressed tar file to the erofs image
+    debug!(
+        "Appending decompressed tar file {:?} to erofs image {:?}",
+        &path, &erofs_path
+    );
+    let mut erofs_file = OpenOptions::new()
```

Comment:
Since you have erofs-common now, you could move this common logic in there, so you are less likely to diverge.

Reply:
Moved common logic into erofs-common - [commit](https://github.com/microsoft/kata-containers/pull/350/commits/ca40d45677db4c4a97dc315cf07e8a59bed97a1c)